cli/cli_app.go

```diff
@@ -377,7 +376,26 @@ func runTerraformCommandIfNoErrors(possibleErrors error, terragruntOptions *opti
 		return err
 	}
 
-	return shell.RunTerraformCommand(terragruntOptions, terragruntOptions.TerraformCliArgs...)
+	var err error = nil
+
+	// Retry the command configurable time with sleep in between
+	for i := 0; i < terragruntOptions.MaxRetryAttempts; i++ {
+		if out, tferr := shell.RunTerraformCommandWithOutput(terragruntOptions, terragruntOptions.TerraformCliArgs...); tferr != nil {
+			err = tferr
+			if isRetryable(out, tferr, terragruntOptions) {
+				terragruntOptions.Logger.Printf("Encountered an error eligible for retrying. Sleeping %v before retrying.\n", terragruntOptions.Sleep)
+				err = tferr
+				time.Sleep(terragruntOptions.Sleep)
+			} else {
+				break
+			}
+		} else {
+			err = nil
+			break
+		}
+	}
+
+	return err
```

Comment:
Based on the above suggestions, if you get here, you should probably return an error that max retries have been exceeded

Reply:
@brikis98 That means that we will not have the error from the terraform execution run_shell_cmd.RunTerraformCommandWithOutput and lose the associated information (which we do for the earlier tries, as well). Is this ok? I know this is the case only when the error is retryable and we've exhausted retries - with any other error we have the information via tferr. Still wanted to double check...